graylog2-server/src/main/java/org/graylog2/indexer/Mapping.java

```diff
@@ -59,78 +56,55 @@ public static PutMappingRequest getPutMappingRequest(final Client client, final
+        final Map<String, String> notAnalyzed = ImmutableMap.of("index", "not_analyzed");
+        final Map<String, Object> defaultAll = ImmutableMap.of(
+                // Match all
+                "match", "*",
+                // Analyze nothing by default
+                "mapping", notAnalyzed);
+        final Map<String, Map<String, Object>> template = ImmutableMap.of("store_generic", defaultAll);
+
+        return ImmutableList.of(template);
     }
 
     /*
      * Enable analyzing for some fields again. Like for message and full_message.
      */
-    // TODO warnings
-    @SuppressWarnings("rawtypes")
-	private static Map partFieldProperties(String analyzer) {
-        final Map<String, Map<?, ?>> properties = Maps.newHashMap();
-
-        properties.put("message", analyzedString(analyzer));
-        properties.put("full_message", analyzedString(analyzer));
-
-        // http://joda-time.sourceforge.net/api-release/org/joda/time/format/DateTimeFormat.html
-        // http://www.elasticsearch.org/guide/reference/mapping/date-format.html
-        properties.put("timestamp", typeTimeWithMillis());
-
-        // to support wildcard searches in source we need to lowercase the content (wildcard search lowercases search term)
-        properties.put("source", analyzedString("analyzer_keyword"));
-
-        return properties;
+    private static Map<String, Map<String, ? extends Serializable>> partFieldProperties(String analyzer) {
+        return ImmutableMap.of(
+                "message", analyzedString(analyzer),
+                "full_message", analyzedString(analyzer),
+                // http://joda-time.sourceforge.net/api-release/org/joda/time/format/DateTimeFormat.html
+                // http://www.elasticsearch.org/guide/reference/mapping/date-format.html
+                "timestamp", typeTimeWithMillis(),
+                // to support wildcard searches in source we need to lowercase the content (wildcard search lowercases search term)
+                "source", analyzedString("analyzer_keyword"));
     }
 
     private static Map<String, String> analyzedString(String analyzer) {
-        final Map<String, String> type = Maps.newHashMap();
-        type.put("index", "analyzed");
-        type.put("type", "string");
-        type.put("analyzer", analyzer);
-
-        return type;
+        return ImmutableMap.of(
+                "index", "analyzed",
+                "type", "string",
+                "analyzer", analyzer);
     }
 
-    private static Map<String, String> typeTimeWithMillis() {
-        final Map<String, String> type = Maps.newHashMap();
-        type.put("type", "date");
-        type.put("format", Tools.ES_DATE_FORMAT);
-
-        return type;
+    private static Map<String, Serializable> typeTimeWithMillis() {
+        return ImmutableMap.<String, Serializable>of(
+                "type", "date",
+                "format", Tools.ES_DATE_FORMAT,
+                "doc_values", true);
```

Comment:
Would be good to make this configurable by the user with the default being 'true'.